### PR TITLE
Options to get NCMC simulation output

### PR DESCRIPTION
**Pull request by sgill2**
This PR adds the following funcitonality:
Adds support for a write_ncmc option for the opt dictionary. The value of write_ncmc is an integer and specifies at what interval the frames are output, so everytime `ncsteps % write_ncmc == 0` a frame is output. This also automatically outputs a frame when a move is performed. 
A openmm dcdreporter outputs the frame to ncmc_outfile, which can also be specifed in opt (the default value is ncmc_output.dcd'). 
Addresses #71

### CODE REVIEW

**Comment by sgill2** [comment]
The test is failing the same reason #70 was failing before–because the openmmtools integrators got changed (and which #70 addresses), so I think this PR should otherwise be fine. It runs the example locally fine.

**Comment by davidlmobley** [comment]
@nathanmlim - can you review?

@sgill2 - did you want to bring the integrator name changes into this PR then?

**Comment by sgill2** [comment]
> did you want to bring the integrator name changes into this PR then?
I'll have to do a little more work to pull just the openmmtool integrator changes in, but sounds good.

**Comment by sgill2** [comment]
Alright! The openmmtool changes seem to have been brought in fine.

**Comment by davidlmobley** [comment]
For future reference, I believe with the appropriate git invocations it's possible to move changes made to one branch into a different branch "magically" so that you could probably have just migrated the integrator-naming related commits over to this branch.

File: blues/simulation.py [hunk 1]
```diff
@@ -77,6 +78,17 @@ def __init__(self, simulations, model, mover, **opt):
...
         else:
             self.verbose = False
+        if 'write_ncmc' in opt:
+            self.write_ncmc = opt['write_ncmc']
+            if 'ncmc_outfile' in opt:
+                self.ncmc_outfile = opt['ncmc_outfile']
+            else: 
+                self.ncmc_outfile = 'ncmc_output.dcd'
+            
+            self.ncmc_reporter = app.dcdreporter.DCDReporter(self.ncmc_outfile, 1)
```
Comment (nathanmlim) [inline blues/simulation.py]: Could we move these into it's own class? That way we don't have reporters being added in two different places. This way we can easily expand it to support writing to other formats if we would like.

Comment (nathanmlim) [inline blues/simulation.py]: Comments in #71

**Comment by sgill2** [comment]
@nathanmlim Could you review this again when you have a moment? I've changed the ncmc reporter to be pretty much consolidated to the simulateNCMC portion of the code.

File: blues/simulation.py [hunk 2]
```diff
@@ -224,6 +224,10 @@ def chooseMove(self):
 
     def simulateNCMC(self, verbose=False):
```
Comment (nathanmlim) [inline blues/simulation.py]: I think we should add a `write_ncmc` parameter here to make it a bit more obvious we can pass that option here. Then everything inside could just be pointing to `write_ncmc`

File: blues/simulation.py [hunk 3]
```diff
@@ -294,19 +298,12 @@ def run(self):
...
-        areporter.report(self.nc_sim, self.nc_sim.context.getState(True,True))
-
-
         #set inital conditions
         self.setStateConditions()
         for n in range(self.nIter):
             self.current_iter = int(n)
             self.setStateConditions()
-            self.simulateNCMC()
+            self.simulateNCMC(verbose=self.verbose)
```
Comment (nathanmlim) [inline blues/simulation.py]: Here, we would add the `write_ncmc=self.write_ncmc`

File: blues/simulation.py (hunk 2 above)
Comment (nathanmlim) [inline blues/simulation.py]: Am I understanding this correctly, in that the reporter will write to a DCD file before and after the move?

Comment (sgill2) [inline blues/simulation.py]: >Am I understanding this correctly, in that the reporter will write to a DCD file before and after the move?

Pretty much, yes. It'll save the frame after the integration step and when a move is performed.

**Pull request merged**